### ISSUE TITLE
Corona texture not adjusted for refraction

### ISSUE THREAD

**Issue by Atque** [Stellarium/stellarium#2264]
Viewing solar eclipses close to the horizon, we see that the corona texture is not affected by atmospheric refraction. See the screenshot below.

![stellarium-117](https://user-images.githubusercontent.com/37028368/153595187-09131a74-b02f-4619-8070-05d38666dca9.png)

Note how the Moon is squeezed while the corona texture is not, creating an ugly artifact.

**Comment by worachate001** [comment]
I'm not sure this can be considered as a fix or not. I just try to make it better by editing the images (not sure this is allowed for image's owner) and adjust the size of corona.

![corona](https://user-images.githubusercontent.com/41257965/154184539-20bf506e-cba2-440a-abc8-94e3201c8cbe.png)

The process make it looks like this for 1957-10-23 total solar eclipse.

![1957-10-23-tse-fix](https://user-images.githubusercontent.com/41257965/154184556-5ec5462a-3bd0-491f-9415-8b6fb179a572.jpg)

**Comment by gzotti** [comment]
No, please don't paint the corona. It is pixel-perfect from the 2008 Mongolian eclipse. What we need to do is adding refraction to the corona texture. Currently it is drawn as simple image that only can be rotated, but not squeezed by-vertex. (I.e. use VertexArray for it.)

**Comment by Atque** [comment]
I just noted that the texture is not aligned with the ecliptic, at least if the eclipse is high in the sky. Look at the July 5 2168 eclipse, and note how the texture rotates as you pan around. It should be fixed.

**Comment by gzotti** [comment]
Yes, when somebody will implement it in a technically fully different way than it is done right now (DSO-like vertex array, not rotated image), this will likewise be fixed.

**Comment by Atque** [comment]
Whoa, the texture does not apply projection corrections either. Look at the recently added Mollweide projection:

![Image](https://github.com/user-attachments/assets/1fd88f86-fa84-44d4-bd75-429c99da6c9a)

**Comment by gzotti** [comment]
>Yes, when somebody will implement it in a technically fully different way than it is done right now (DSO-like vertex array, not rotated image), this will likewise be fixed.
